src/app/cli/interpreter.cpp

```diff
@@ -1432,20 +1437,47 @@ Interpreter::Value Interpreter::ProcessBr(const Expression &aExpr)
         nlohmann::json                            baJson;
         char                                      mdnsSendBuffer[kMdnsBufferSize];
 
-        if (mContext.mCommandKeys.size() == 2 && mContext.mCommandKeys[0] == "--timeout")
+        for (auto it = mContext.mCommandKeys.begin(); it != mContext.mCommandKeys.end(); ++it)
         {
-            try
+            if (*it == "--timeout")
             {
-                scanTimeout = stol(mContext.mCommandKeys[1]);
-            } catch (...)
+                if (++it != mContext.mCommandKeys.end())
+                {
+                    try
+                    {
+                        scanTimeout = stol(*it);
+                    } catch (...)
+                    {
+                        ExitNow(value = ERROR_INVALID_ARGS("Imparsable timeout value '{}'", *it));
+                    }
+                }
+                else
+                {
+                    ExitNow(value = ERROR_INVALID_ARGS("Missing {} value", *--it));
+                }
+            }
+            else if (*it == "--netif")
             {
-                ExitNow(value = ERROR_INVALID_ARGS("Imparsable timeout value '{}'", aExpr[3]));
+                if (++it != mContext.mCommandKeys.end())
+                {
+                    netIf = *it;
+                }
+                else
+                {
+                    ExitNow(value = ERROR_INVALID_ARGS("Missing {} value", *--it));
+                }
             }
         }
```

Comment:
I prefer your previous implementation which use `std::find` to get the options you care about, which is more clearer and straight forward :)

The current implementation here needs to be very careful with the iterators (the ++, --) and I believe you spent some time to get it correct. So it's really harder to maintain this version.

Please consider using the `std::find` version.

Reply:
I agree that `std::find` is easier to understand and use. However, the current method might be more efficient. It only iterates through `mContext.mCommandKeys` once, whereas `std::find` might need to iterate multiple times.

Reply:
Performance doesn't matter in a CLI tool like this :)

Reply:
Okay, Let me update it again

Reply:
updated, Thanks.